File: sdk/python/llm_engine/ai_lifecycle.py
"""Python SDK Client for Enterprise AI Lifecycle Platform (Phase 5.33)."""

from typing import Dict, Any, Optional, List


class AILifecycleClient:
    """SDK Client for interacting with Enterprise AI Lifecycle Platform API."""

    def __init__(self, base_client: Any = None, base_url: str = "") -> None:
        self._client = base_client
        self.base_url = base_url

    def register_asset(self, tenant_id: str, name: str, asset_type: str = "MODEL", description: str = "") -> Dict[str, Any]:
        return self._client.post(f"/v1/ai-lifecycle/assets?tenant_id={tenant_id}", json={"name": name, "asset_type": asset_type, "description": description})

    def register_dataset(self, tenant_id: str, name: str, classification: str = "INTERNAL") -> Dict[str, Any]:
        return self._client.post(f"/v1/ai-lifecycle/datasets?tenant_id={tenant_id}", json={"name": name, "classification": classification})

    def register_model(self, tenant_id: str, name: str, model_type: str = "LLM", framework: str = "TRANSFORMERS") -> Dict[str, Any]:
        return self._client.post(f"/v1/ai-lifecycle/models?tenant_id={tenant_id}", json={"name": name, "model_type": model_type, "framework": framework})

    def register_agent(self, tenant_id: str, name: str, agent_type: str = "TASK_AGENT", autonomy_level: str = "HUMAN_APPROVED") -> Dict[str, Any]:
        return self._client.post(f"/v1/ai-lifecycle/agents?tenant_id={tenant_id}", json={"name": name, "agent_type": agent_type, "autonomy_level": autonomy_level})

    def run_evaluation(self, tenant_id: str, target_asset_id: str, suite_id: str, overall_passed: bool = True) -> Dict[str, Any]:
        return self._client.post(f"/v1/ai-lifecycle/evaluations?tenant_id={tenant_id}", json={"target_asset_id": target_asset_id, "suite_id": suite_id, "overall_passed": overall_passed})

    def create_gate(self, tenant_id: str, name: str, gate_type: str = "SECURITY", is_hard_gate: bool = True) -> Dict[str, Any]:
        return self._client.post(f"/v1/ai-lifecycle/gates?tenant_id={tenant_id}", json={"name": name, "gate_type": gate_type, "is_hard_gate": is_hard_gate})

    def request_promotion(self, tenant_id: str, asset_id: str, target: str = "STAGING", is_high_risk: bool = False) -> Dict[str, Any]:
        return self._client.post(f"/v1/ai-lifecycle/promotions?tenant_id={tenant_id}", json={"asset_id": asset_id, "target": target, "is_high_risk": is_high_risk})

    def create_release(self, tenant_id: str, title: str, asset_id: str, version: str = "1.0.0", risk_level: str = "MEDIUM") -> Dict[str, Any]:
        return self._client.post(f"/v1/ai-lifecycle/releases?tenant_id={tenant_id}", json={"title": title, "asset_id": asset_id, "version": version, "risk_level": risk_level})

    def detect_drift(self, tenant_id: str, asset_id: str, drift_type: str = "PERFORMANCE_DRIFT", severity: str = "HIGH") -> Dict[str, Any]:
        return self._client.post(f"/v1/ai-lifecycle/drift?tenant_id={tenant_id}", json={"asset_id": asset_id, "drift_type": drift_type, "severity": severity})

    def request_rollback(self, tenant_id: str, asset_id: str, target_version: str = "1.0.0", reason: str = "Performance degradation") -> Dict[str, Any]:
        return self._client.post(f"/v1/ai-lifecycle/rollbacks?tenant_id={tenant_id}", json={"asset_id": asset_id, "target_version": target_version, "reason": reason})

    def request_retirement(self, tenant_id: str, asset_id: str, reason: str = "DEPRECATED") -> Dict[str, Any]:
        return self._client.post(f"/v1/ai-lifecycle/retirement?tenant_id={tenant_id}", json={"asset_id": asset_id, "reason": reason})

    def get_analytics_report(self, tenant_id: str) -> Dict[str, Any]:
        return self._client.get(f"/v1/ai-lifecycle/analytics?tenant_id={tenant_id}")


class AILifecyclePlatformClient(AILifecycleClient):
    pass

File: sdk/typescript/src/ai_lifecycle.ts
/**
 * TypeScript SDK Client for Enterprise AI Lifecycle Platform (Phase 5.33).
 */

export class AILifecycleClient {
  private client: any;

  constructor(client: any) {
    this.client = client;
  }

  async registerAsset(tenantId: string, name: string, assetType: string = "MODEL", description: string = ""): Promise<any> {
    return this.client.post(`/v1/ai-lifecycle/assets?tenant_id=${tenantId}`, { name, asset_type: assetType, description });
  }

  async registerDataset(tenantId: string, name: string, classification: string = "INTERNAL"): Promise<any> {
    return this.client.post(`/v1/ai-lifecycle/datasets?tenant_id=${tenantId}`, { name, classification });
  }

  async registerModel(tenantId: string, name: string, modelType: string = "LLM", framework: string = "TRANSFORMERS"): Promise<any> {
    return this.client.post(`/v1/ai-lifecycle/models?tenant_id=${tenantId}`, { name, model_type: modelType, framework });
  }

  async registerAgent(tenantId: string, name: string, agentType: string = "TASK_AGENT", autonomyLevel: string = "HUMAN_APPROVED"): Promise<any> {
    return this.client.post(`/v1/ai-lifecycle/agents?tenant_id=${tenantId}`, { name, agent_type: agentType, autonomy_level: autonomyLevel });
  }

  async runEvaluation(tenantId: string, targetAssetId: string, suiteId: string, overallPassed: boolean = true): Promise<any> {
    return this.client.post(`/v1/ai-lifecycle/evaluations?tenant_id=${tenantId}`, { target_asset_id: targetAssetId, suite_id: suiteId, overall_passed: overallPassed });
  }

  async createGate(tenantId: string, name: string, gateType: string = "SECURITY", isHardGate: boolean = true): Promise<any> {
    return this.client.post(`/v1/ai-lifecycle/gates?tenant_id=${tenantId}`, { name, gate_type: gateType, is_hard_gate: isHardGate });
  }

  async requestPromotion(tenantId: string, assetId: string, target: string = "STAGING", isHighRisk: boolean = false): Promise<any> {
    return this.client.post(`/v1/ai-lifecycle/promotions?tenant_id=${tenantId}`, { asset_id: assetId, target, is_high_risk: isHighRisk });
  }

  async createRelease(tenantId: string, title: string, assetId: string, version: string = "1.0.0", riskLevel: string = "MEDIUM"): Promise<any> {
    return this.client.post(`/v1/ai-lifecycle/releases?tenant_id=${tenantId}`, { title, asset_id: assetId, version, risk_level: riskLevel });
  }

  async detectDrift(tenantId: string, assetId: string, driftType: string = "PERFORMANCE_DRIFT", severity: string = "HIGH"): Promise<any> {
    return this.client.post(`/v1/ai-lifecycle/drift?tenant_id=${tenantId}`, { asset_id: assetId, drift_type: driftType, severity });
  }

  async requestRollback(tenantId: string, assetId: string, targetVersion: string = "1.0.0", reason: string = "Performance degradation"): Promise<any> {
    return this.client.post(`/v1/ai-lifecycle/rollbacks?tenant_id=${tenantId}`, { asset_id: assetId, target_version: targetVersion, reason });
  }

  async requestRetirement(tenantId: string, assetId: string, reason: string = "DEPRECATED"): Promise<any> {
    return this.client.post(`/v1/ai-lifecycle/retirement?tenant_id=${tenantId}`, { asset_id: assetId, reason });
  }

  async getAnalyticsReport(tenantId: string): Promise<any> {
    return this.client.get(`/v1/ai-lifecycle/analytics?tenant_id=${tenantId}`);
  }
}
